script.js

```diff
@@ -1,4 +1,943 @@
+            playerAisplay +
+            " vs " +
+            "Player " +
+            playerBdisplay +
+            " ----> " +
+            winnerThisRound +
+            " wins"
+        );
+      }
+      console.log("Round: " + (playerIndexNmbr - 1));
+      console.log("Incumbent = Player " + incumbentIndex);
+      console.log("Current Challenger = Player " + playerIndexNmbr);
+      console.log("============================");
+    }
+
+    // For Output template
+    var message2 = `Dear Players, please find your numbers below:
+                    <br><br>Combined Numbers of each player: <br>${leaderBoardArray}
+                    <br><br> Winner of KnockOut: ${winnerThisRound}.
+                    <br><br> Round History: ${roundHistoryDisplay}
+                        <br><i>in event of a draw, previous winner always wins.</i>
+                    <br><br>A new round will begin and all scores will reset on clicking "Submit" 
+                    <br><br>If you wish to change "Highest/Lowest" wins game logic, please refresh the page. `;
+  } else {
+    message2 = `${playerName}, please only:<br> 1) click the submit button, or<br> 2) type in "lowest", or, <br>3) type in "highest".`; //for catching error in inputs
+  }
+
+  //for restarting next game.
+  playerIndexNmbr = 0;
+  incumbentIndex = 0;
+  playerName = "";
+  gameMode = `waiting for input`;
+  return gameLogicDisplay + message2;
+};
+
+//GLOBAL DEFINITIONS for MAIN FUCNTION
+var gameMode = "waiting for input";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  var waitingForInputMsg = ` HOME OUTPUT SCREEN:
+                           <br><br>Game Logic: ${gameLogic.toUpperCase()} combined number wins
+                           <br><br>Please enter the number of players, 
+                           <br><br>or type "highest" or "lowest" to set game logic.`;
+  if (
+    gameMode == `waiting for input` &&
+    playerName == "" &&
+    (input.toLowerCase() == "highest" || input == "") // if user keys in "highest" or not, logic = highest wins
+  ) {
+    gameLogic = "highest";
+    myOutputValue = waitingForInputMsg;
+    gameMode = "getNumberOfPlayers";
+  } else if (
+    gameMode == `waiting for input` &&
+    playerName == "" &&
+    input.toLowerCase() == "lowest"
+  ) {
+    gameLogic = "lowest";
+    myOutputValue = waitingForInputMsg;
+    gameMode = "getNumberOfPlayers";
+    gameLogic = "lowest";
+  } else if (gameMode == `getNumberOfPlayers` && playerName == "") {
+    myOutputValue = getNumberOfPlayers(input);
+  } else if (gameMode == `inputQtyOfDice`) {
+    myOutputValue = inputQtyOfDice(input);
+  } else if (gameMode == `setMultiplayerOrKnockOut`) {
+    myOutputValue = setMultiplayerOrKnockOut(input);
+  } else if (gameMode == `arrangeDiceMultiplayer`) {
+    myOutputValue = arrangeDiceMultiplayer(input);
+  } else if (gameMode == `arrangeDiceKnockOut`) {
+    myOutputValue = arrangeDiceKnockOut(input);
+  }
+
```

Comment:
You may want to look at using a switch case for a situation where you have multiple else ifs for the same condition:
https://www.w3schools.com/js/js_switch.asp